server/handlers/result_handlers.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional
+import logging
+import json
+from abc import abstractmethod
+from pydantic import computed_field, ConfigDict, model_validator
+
+from adala.utils.registry import BaseModelInRegistry
+from label_studio_sdk import Client
+
+
+logger = logging.getLogger(__name__)
+
+
+class ResultHandler(BaseModelInRegistry):
+    """
+    Abstract base class for a result handler.
+    This is a callable that is instantiated in `/submit-streaming` with any arguments that are needed, and then is called on each batch of results when it is finished being processed by the Agent (it consumes from the Kafka topic that the Agent produces to).
+
+    It can be used as a connector to load results into a file or external service. If a ResultHandler is not used, the results will be discarded.
+
+    Subclasses must implement the `__call__` method.
+
+    BaseModelInRegistry is a utility class that allows polymorphic instantiation of ResultHandlers through passing in the class name as the "type" field in the request, for example:
```

Comment:
these docs are super helpful!

I have one tiny nit that you can take or leave: imo just saying something like this could be clearer:
> The `BaseModelInRegistry` base class implements a factory pattern, allowing the `type` parameter to specify which subclass of `ResultHandler` to instantiate. For example:

Not everyone necessarily knows what a factory pattern is, but it's very Google-able and more specific than just "polymorphic instantiation".

Reply:
ah yes that's the term I was looking for :slightly_smiling_face: